src/eventHandlers/WorkerInitHandler.ts

```diff
@@ -34,35 +33,12 @@ export class WorkerInitHandler extends EventHandler<'workerInitRequest', 'worker
 
         logColdStartWarning(channel);
         const functionAppDirectory = nonNullProp(msg, 'functionAppDirectory');

```

Comment:
I did not test specialization locally, but judging from host code [here](https://github.com/Azure/azure-functions-host/blob/1db1782f1bda1718215ce7337e89ebde32515325/src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs#L232) and @pragnagopa 's comments on the thread we had about worker specialization: "In PlaceholderMode,  Host starts with dummy app content , script path points to a local directory that has a .csx function," I doubt that this value should be `undefined` in normal operations

Reply:
Should we at least log some kind of warning in the case where `msg.functionAppDirectory` is undefined? Because that means our app startup logic will not run on a `workerInitRequest` and that might lead to some unexpected behavior, and perhaps help with servicing too

Reply:
I would honestly prefer if `msg.functionAppDirectory` was `undefined` instead of pointing to dummy app content, because I don't necessarily trust the dummy app content to stay the same forever. So no, I would not call it "unexpected behavior" when it's my preferred behavior (for the case of placeholders).

Keep in mind that too many logs or misleading logs can make servicing more difficult, so more logs does not always equal better. If you feel strongly that this needs a log, I would be okay with a compromise where you add a log at the "Information" level, which is easier to filter and does not construe this as a problem

Reply:
No that makes sense, I'm okay with not having the log. It is more unexpected behavior when starting the worker in "normal" rather than "placeholder" mode. If anything, I think it would be more unexpected for the `functionAppDirectory` to be `undefined` in the `functionEnvironmentReloadRequest` when specializing, and I see that there, we don't have a log if it is `undefined`, so since there's precedent there, I'm okay with that.